backend/regalloc/regalloc_gi_utils.ml

```diff
@@ -456,27 +466,30 @@ module Hardware_registers = struct
        a stack slot *)
     SpillCosts.for_reg costs reg
 
+  exception Found
+
+  let exists_assigned (tbl : Hardware_register.assigned Reg.Tbl.t)
+      ~(f : Hardware_register.assigned -> bool) =
+    try
+      Reg.Tbl.iter (fun _reg assigned -> if f assigned then raise Found) tbl;
+      false
+    with Found -> true
+
   let overlap (hardware_reg : Hardware_register.t) (interval : Interval.t) :
       bool =
     if debug
     then (
       log "considering %a" Hardware_register.print_location
         hardware_reg.location;
       indent ());
-    let overlap_hard : bool = Interval.overlap interval hardware_reg.interval in
-    let overlap_assigned =
-      List.exists hardware_reg.assigned
-        ~f:(fun
-             { Hardware_register.pseudo_reg = _; interval = itv; evictable = _ }
-           -> Interval.overlap itv interval)
-    in
-    let overlap = overlap_hard || overlap_assigned in
-    if debug
-    then (
-      log "overlap=%B (hard=%B, assigned=%B)" overlap overlap_hard
```

Comment:
Is this log intentionally deleted?

Reply:
Yes, it was not really useful,
and it is wasteful to evaluate
both sub-conditions every
time.